### ISSUE TITLE
Populate a list of WeaponHash(es) from memory during startup

### ISSUE THREAD

**Issue by justalemon** [crosire/scripthookvdotnet#966]
I know that is possible to get the [Vehicle, Ped and Weapon Model hashes](https://github.com/crosire/scripthookvdotnet/blob/2a2ad048ccc897f17aba03c780f2df2104b83980/source/core/NativeMemory.cs#L376-L379), but I would really like to a list of weapon from memory.

You might be thinking, why? The WeaponHash enum not always contains all of the weapons after an update (see #945) and not being a good solution for addon weapons.

In other words, instead of having to do something like:

```cs
(WeaponHash[])Enum.GetValues(typeof(WeaponHash));  // Only the weapons in the Enum
```

Have a solution like this:

```cs
Weapon.GetAllHashes();  // From memory, including those not present in the Enum
```

**Comment by kagikn** [comment]
alloc8or said like this in 5Mods Discord
>weapon hashes are stored in CWeaponInfo (at 0x10)
>so, uh, one method would be to simply access the $``rage::atArray<CWeaponInfo*>`` at ``48 8B 05 ? ? ? ? 41 8B 1E``
The array contains `CWeaponInfo` and `CAmmoInfo` elements.
The elements of the array is sorted in as sending order of name hash (natives that uses weapon hashes use the trick).
The problem is, how can we skip `CAmmoInfo` so we can create an array that only contains weapon hashes?

**Comment by kagikn** [comment]
Oh wait, [Parik27 found a way to get hash of the most concrete class name of `CItemInfo`](https://github.com/Parik27/V.Rainbomizer/blob/825f24aaebc17fd62ace438cf49c0bda820b3459/lib/CItemInfo.hh#L19), so we can get weapon hashes from only `CWeaponInfo` like this;
```
    var vtable = *(ulong*)cAmmoInfoAddress;
    var getClassIdOfCAmmoInfo= GetDelegateForFunctionPointer<GetClassIdOfCAmmoInfoDelegate>(new IntPtr((long)(vtable + 0x10)));

    int outVal = 0;
    var classNameHash = getClassIdOfCAmmoInfo(&outVal);

    const uint cWeaponInfoNameHash = 0x861905B4;
    if (classNameHash == cWeaponInfoNameHash)
    {
        returnList.Add(*(int*)(cAmmoInfoAddress + 0x10));
    }
```

We should also provide a method for the weapon hashes that peds can carry. The array contains weapon hashes that are not for peds and vehicle weapons (e.g. `WEAPON_FALL`).
